### PR TITLE
Adds newlines to log statements

### PR DESCRIPTION
**Pull request by delambo**
This is pretty small. I added some newlines to the `start`, `end`, and `error` logs. The `dev` command looks like the following after a hot reload:

```

ℹ️  Using kyt config at /Users/202586/Sites/test.nytimes.com/kyt-starter-universal/kyt.config.js

🔥  Starting development build...

👍  Cleaned ./build
👍  Client webpack configuration compiled
👍  Server webpack configuration compiled
👍  Client build successful
👍  Setup React Hot Loader
👍  Client assets serving from http://localhost:3001/assets/
webpack built 7741101ff391e9a5c1c4 in 4316ms
👍  Server build successful
👍  Server running at: http://localhost:3000/

✅  Development started

webpack building...
👍  Client build successful
webpack built a38681c47339c9e21496 in 1188ms
👍  Server build successful

✅  Development server restarted

```

### CODE REVIEW

**Comment by tizmagik** [comment]
What if instead of hard-coding the new-line char to only `start` `end` and `error` you handled it generically for all console logs to consume a log that starts with a new-line char and then pass along the rest of the call as normal? E.g.

```js
const write = (status, text, verbose) => {
    if (text.startsWith('\n')) {
        logger.log('');
        write(status, text.slice(1), verbose);
        return;
    }

    // ...
}
```

And that way, when some new-line "padding" is desired, you can just log like so:

```js
logger.end('\nDevelopment server restarted');
```

**Comment by delambo** [comment]
My idea was to group tasks and to always separate the important boundary log statements like start and end. In general all logs would follow the following pattern:

```
Start

task 1
task 2
task 3

End
```

For long-running commands like `dev`, the restart tasks would be grouped and followed with an end log:

```
Start

task 1
task 2

End

restarting...
task 1
task 2

Error

restarting...
task 1
task 2

End

...
```

**Comment by tizmagik** [comment]
I see, cool! I guess that makes sense for `start` and `end`, but why for `error`? `start` and `end` are not part of the canonical `console` API so it makes sense to add it as custom to the `logger` API, but I'm not sure that it makes sense to do that to `error`? What do you think?

File: cli/__tests__/logger.test.js [hunk 1]
```diff
@@ -9,21 +9,28 @@ describe('logger', () => {
...
 
+  const newlineMethods = ['start', 'end', 'error'];
+
   [...oneArgMethods, ...twoArgMethods].forEach(({ method, expectedPrefix }) => {
     it(`logger.${method}: simple usage`, () => {
       logger[method]('here is some text');
-      expect(global.console.log.mock.calls).toEqual([[`${expectedPrefix}here is some text`]]);
+      if (newlineMethods.indexOf(method) > -1) {
+        expect(global.console.log.mock.calls)
+          .toEqual([[`${expectedPrefix}here is some text`], []]);
```
Comment (tizmagik) [inline cli/__tests__/logger.test.js]: Could this be written as this?:

```js
expect(global.console.log).toBeCalledWith(`${expectedPrefix}here is some text`);
```

File: cli/__tests__/logger.test.js [hunk 2]
```diff
@@ -9,21 +9,28 @@ describe('logger', () => {
...
+      } else {
+        expect(global.console.log.mock.calls).toEqual([[`${expectedPrefix}here is some text`]]);
```
Comment (tizmagik) [inline cli/__tests__/logger.test.js]: I guess the same `toBeCalledWith()` could be used here as well and for the rest of this test, right?

Comment (jaredmcdonald) [inline cli/__tests__/logger.test.js]: @tizmagik the difference here is that this also verifies it was called exactly once, with only that one argument... could add some more assertions to get that too, maybe this is too opaque

Comment (tizmagik) [inline cli/__tests__/logger.test.js]: Yea, I think it's a bit too opaque, I think something like this is more descriptive:

```js
expect(global.console.log).toBeCalledWith(`${expectedPrefix}here is some text`);
expect(global.console.log.mock.calls).toEqual(1);
```

Or, when we update to `Jest v16`:

```js
expect(global.console.log).toBeCalledWith(`${expectedPrefix}here is some text`);
expect(global.console.log).toHaveBeenCalledTimes(1);
```

Comment (delambo) [inline cli/__tests__/logger.test.js]: Fixed the tests up.

**Comment by tizmagik** [comment]
Looks good! I still think we shouldn't add newlines for `logger.error()` because that would be unexpected, but I'll leave that up to you.

**Pull request merged**